rxgroups-android/src/main/java/com/airbnb/rxgroups/android/GroupLifecycleManager.java

```diff
@@ -0,0 +1,172 @@
+package com.airbnb.rxgroups.android;
+
+import android.app.Activity;
+import android.app.Fragment;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.airbnb.rxgroups.ObservableGroup;
+import com.airbnb.rxgroups.ObservableManager;
+
+import javax.annotation.Nullable;
+
+import rx.Observable;
+import rx.Observer;
+import rx.Subscription;
+import rx.functions.Action1;
+import rx.functions.Func1;
+import rx.subscriptions.CompositeSubscription;
+
+public class GroupLifecycleManager {
+  private static final String TAG = "GroupLifecycleManager";
+  private static final String KEY_MANAGER_HASHCODE = "KEY_MANAGER_HASHCODE";
+  private static final String KEY_GROUP_ID = "KEY_GROUP_ID";
+
+  private final CompositeSubscription pendingSubscriptions = new CompositeSubscription();
+  private final LifecycleResubscription resubscription;
+  private final ObservableManager observableManager;
+  private final ObservableTagFactory tagFactory;
+  private ObservableGroup group;
+  private boolean hasSavedState;
+
+  public GroupLifecycleManager(ObservableManager observableManager,
+      ObservableTagFactory tagFactory, LifecycleResubscription resubscription) {
+    this.observableManager = observableManager;
+    this.tagFactory = tagFactory;
+    this.resubscription = resubscription;
+    group = observableManager.newGroup();
+    group.lock();
+  }
+
+  public GroupLifecycleManager(ObservableManager observableManager,
+      ObservableTagFactory tagFactory) {
+    this(observableManager, tagFactory, new LifecycleResubscription());
+  }
+
+  public void saveState(Bundle outState) {
+    hasSavedState = true;
+    outState.putInt(KEY_MANAGER_HASHCODE, observableManager.hashCode());
+    outState.putLong(KEY_GROUP_ID, group.id());
+  }
+
+  public void restoreState(@Nullable Bundle savedState, @Nullable Object target) {
+    // Only need to resubscribe if restoring from saved state
+    boolean shouldResubscribe = target != null && savedState != null;
+
+    if (savedState != null) {
+      int hashCode = savedState.getInt(KEY_MANAGER_HASHCODE);
+      long groupId = savedState.getLong(KEY_GROUP_ID);
+
+      if (hashCode != observableManager.hashCode()) {
```

Comment:
We should add documentation about why we save the hashcode and check for differences